### PR TITLE
Make built-in indexer work for retrieval client

### PR DESCRIPTION
**Pull request by wmb-software-consulting**
## Why are these changes needed?

Context issue:
- Users previously only required a retrieval client with a built-in indexer.
- Current changes necessitate the addition of a graph node, adding complexity for users.

Solution in this PR: 
- Prioritize making the built-in indexer effective instead of relying on the graph node.

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [x] Unit tests
   - [x] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jianoaix — COMMENTED** [review]
Can you summarize the issues that this is fixing?

File: clients/retrieval_client.go [hunk 1]
```diff
@@ -35,20 +38,32 @@ var _ RetrievalClient = (*retrievalClient)(nil)
...
-) *retrievalClient {
+) (*retrievalClient, error) {
+	indexedState, err := coreindexer.NewIndexedChainState(
+		chainState,
+		indexer,
+	)
+	if err != nil {
+		return nil, err
+	}
+	err = indexedState.Start(context.Background())
```
Comment (ian-shim) [inline clients/retrieval_client.go:55]: Do we need to `Start` the indexedState from the retrieval client? Currently, it's left as the responsibility of caller to provide an indexedState that's actively indexing. 

Even if we do, I don't think this should be called at initialization. If we need to explicitly start a goroutine, it should be separated out as an explicit method

File: disperser/cmd/batcher/main.go [hunk 2]
```diff
@@ -121,9 +123,46 @@ func RunBatcher(ctx *cli.Context) error {
...
+			{
+				Acc:      coreindexer.NewOperatorSocketsAccumulator(logger),
+				Filterer: socketsFilterer,
+				Status:   indexer.Good,
+			},
+		}
+
+		var (
+			upgrader    = &coreindexer.Upgrader{}
+			headerStore = inmemstore.NewHeaderStore()
```
[the remark this answers is not in the document]
Comment (wmb-software-consulting) [reply disperser/cmd/batcher/main.go:153]: As far as I know, we are using inmem because the levelDB has an resolved issue
 https://github.com/Layr-Labs/eigenda-internal/issues/77

[the remark this answers is not in the document]
Comment (wmb-software-consulting) [reply disperser/cmd/batcher/main.go:153]: Regarding the why and constraints,  I think @mooselumph  would be the best person to answer that

File: core/indexer/indexer.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	"fmt"
+
+	dacommon "github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/indexer"
+	indexereth "github.com/Layr-Labs/eigenda/indexer/eth"
+	inmemstore "github.com/Layr-Labs/eigenda/indexer/inmem"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+func SetupNewIndexer(
```
[the remark this answers is not in the document]
Comment (jianoaix) [reply core/indexer/indexer.go:13]: Yes, but this function is call that indexer.New(), so this will be better named as `CreateNewIndexer`.

**Review by jianoaix — COMMENTED** [review]
lgtm

**Review by jianoaix — APPROVED** [review]
